indica/equilibrium.py

```diff
@@ -110,42 +114,47 @@ def __init__(
-            # fitter = SplineFit(lower_bound=0.0, sess=sess)
-            #
-            # T_e_sep = concat(
-            #     [
-            #         fitter(separatrix, t, T_e_with_rho.sel(offset=offset))[0]
-            #         for offset in offsets
-            #     ],
-            #     offsets,
-            # )
-
-            # square_residuals = (T_e_sep - 100.0) ** 2
-            # best_fits = square_residuals.offset[square_residuals.argmin(dim="offset")]
-            # overall_best = best_fits.mean()
-            # fluxes = rhos.sel(offset=overall_best, method="nearest")
-            # offset = float(fluxes.offset)
-            # offset, accept = offset_picker(offset, T_e, fluxes, best_fits)
-            # while not accept:
-            #     fluxes = self.rho.interp(
-            #         t=T_e.coords["t"], method="nearest"
-            #     ).indica.interp2d(
-            #         R=T_e.coords["index"] - offset,
-            #         z=T_e.coords["index_z_offset"] - z_shift,
-            #         method="cubic",
-            #         assume_sorted=True,
-            #     )
-            #     offset, accept = offset_picker(offset, T_e, fluxes, best_fits)
-            self.R_offset = R_shift
+
+            thetas = np.arctan2(
+                T_e.coords["index_z_offset"] + z_shift - zmag,
+                T_e.coords["index"] + offsets - Rmag,
+            )
+            T_e_with_rho = T_e.expand_dims(
+                cast(Dict[Hashable, Any], {"offset": offsets})
+            ).assign_coords(rho_poloidal=rhos, theta=thetas)
+            fitter = SplineFit(lower_bound=0.0, sess=sess)
+
+            T_e_sep = concat(
+                [
+                    fitter(separatrix, t, T_e_with_rho.sel(offset=offset))[0]
+                    for offset in offsets
+                ],
+                offsets,
+            )
+
+            square_residuals = (T_e_sep - 100.0) ** 2
+            best_fits = square_residuals.offset[square_residuals.argmin(dim="offset")]
+            overall_best = best_fits.mean()
+            fluxes = rhos.sel(offset=overall_best, method="nearest")
+            offset = float(fluxes.offset)
+            offset, accept = offset_picker(offset, T_e, fluxes, best_fits)
+            while not accept:
+                fluxes = self.rho.interp(
+                    t=T_e.coords["t"], method="nearest"
+                ).indica.interp2d(
+                    R=T_e.coords["index"] - offset,
+                    z=T_e.coords["index_z_offset"] - z_shift,
+                    method="cubic",
+                    assume_sorted=True,
+                )
+                offset, accept = offset_picker(offset, T_e, fluxes, best_fits)
+
+                # For escaping the while loop while unit testing. Otherwise
+                # with offset_picker = MagicMock(return_value=(0.02, False)),
+                # then this while loop will not end.
+                if unit_testing:
```

Comment:
Instead of adding a specific `unit_testing`, could replace offset with
```python
offset = MagicMock(side_effect=[(0.02, True), (0.02, False)])
```
and it will iterate through those return tuples per call, going into the `while not accept` after first call but exiting after the second